src/PdfReader/index.tsx

```diff
@@ -30,41 +51,121 @@ function pdfReducer(state: PdfState, action: PdfReaderAction): PdfState {
         ...state,
         isScrolling: action.isScrolling,
       };
+
+    case 'SET_NUMPAGES':
+      return {
+        ...state,
+        numPages: action.numPages,
+      };
+
+    case 'SET_PAGENUM':
+      return {
+        ...state,
+        pageNumber: action.pageNum,
+      };
+  }
+}
+async function fetchPdf<ExpectedResponse extends any = any>(url: string) {
+  const response = await fetch(url, { mode: 'cors' });
+  const array = new Uint8Array(await response.arrayBuffer());
+
+  if (!response.ok) {
+    throw new Error('Response not Ok for URL: ' + url);
   }
+  return array as ExpectedResponse;
 }
 
 export default function usePdfReader(args: ReaderArguments): ReaderReturn {
-  const { webpubManifestUrl, manifest } = args ?? {};
-  const [state] = React.useReducer(pdfReducer, {
+  const { webpubManifestUrl, manifest, proxyUrl = '' } = args ?? {};
+
+  const [state, dispatch] = React.useReducer(pdfReducer, {
     type: 'PDF',
+    loadSuccess: false,
     colorMode: 'day',
     isScrolling: false,
     fontSize: 16,
     fontFamily: 'sans-serif',
+    resourceIndex: 0,
+    data: null,
+    pageNumber: 1,
+    numPages: 0,
   });
 
+  const loadResource = async (
+    manifest: WebpubManifest,
+    resourceIndex: number,
+    proxyUrl?: string
+  ) => {
+    // Generate the resource URL using the proxy
+    const resource: string =
+      proxyUrl + encodeURI(manifest.readingOrder![resourceIndex].href);
+
+    return await fetchPdf(resource);
+  };
+
   // initialize the pdf reader
   React.useEffect(() => {
+    async function setPdfResource(manifest: WebpubManifest, proxyUrl: string) {
+      const data = await loadResource(manifest, 0, proxyUrl);
+      dispatch({ type: 'SET_DATA', data: { data } });
+      dispatch({ type: 'SET_RESOURCEINDEX', index: 0 });
+    }
     // bail out if there is not manifest passed in,
     // that indicates that this format is inactive
     if (!manifest) return;
-    // here initialize reader however u do
-    // ...
-  }, [manifest]);
+    // throw an error on a badly formed manifest
+    if (!manifest.readingOrder || !manifest.readingOrder.length) {
+      throw new Error('Manifest has no Reading Order');
+    }
```

Comment:
Why are we explicitly checking the ReadingOrder?

Reply:
the `ReadingOrder` is the part of the webpub that lists the resources in the order they're supposed to be read.  If there's no reading order, there's something wrong with the manifest.